include/irr/asset/filters/CSwizzleAndConvertImageFilter.h

```diff
@@ -150,11 +149,18 @@ class CSwizzleAndConvertImageFilter : public CImageFilter<CSwizzleAndConvertImag
 			typedef std::conditional<asset::isIntegerFormat<inFormat>(), uint64_t, double>::type decodeBufferType;
 			typedef std::conditional<asset::isIntegerFormat<outFormat>(), uint64_t, double>::type encodeBufferType;
 
-			auto perOutputRegion = [&blockDims,&state](const CMatchedSizeInOutImageFilterCommon::CommonExecuteData& commonExecuteData, CBasicImageFilterCommon::clip_region_functor_t& clip) -> bool
+			struct
+			{
+				using BLUE_NOISE_DITHER = asset::CPrecomputedDither;
+				BLUE_NOISE_DITHER dithering;
+				BLUE_NOISE_DITHER::state_type state;  /* how should I use asset manager there in constructor? i dont have an access */
+			} blueNoise;
```

Comment:
make the dither type a template parameter and keep a `Dither*` pointer to the object in the state...

then obvisously you create the dither before you call the filter  ;)

Reply:
corrected